Improving performance

Jeremi-Maciejewski opened the pull request:
Switch from Refclasses to R6.

Benchmark on reading 1Gb file:
```
Memory values, best of 5 tests (per class type).
Memory used is in Mb.
type     Prior    After    Max(diff)    ProcTime    SysTime

Reference    50.8    2978.4    3834.8(3784)    137.552    137.552
R6      51.7    2864.8    3751.2(3700)    29.788    29.788
```

As you can see that is ~4 times faster

Commit:
Edited R/mzQC.R:
Replaced reference classes with R6 classes.

Commit:
Simplified (removed some obsolete features) R6 classes to improve performance; fixed creation of asJSON methods.

Commit:
Fixed creation of asJSON methods, again.

Commit:
Removed some dead (commented out) code.

Commit:
Fixed asJSON methods AGAIN; Created documentation for each R6 class's methods; Adjusted unit tests not to falsely detect modified code as not working.

cbielow commented on R/mzQC.R:
```diff
@@ -1,1010 +1,1310 @@
+    
+    ## Methods
+    
+    #' @description
+    #' Constructor
+    #' @param date Optional POSIXct value for initialization of field datetime. Defaults to Sys.time()
```
todo: check this produces proper docs

cbielow commented on R/mzQC.R:
```diff
@@ -1,1010 +1,1310 @@
+    
+    #' @description
+    #' Constructor
+    #' @param accession Optional string value for initialization of field accession
```
Delete `Optional ` if required (global, i.e. in this file).
Bonus: scratch (` = NA_character_`) where appropriate

cbielow reviewed (changes requested):
nice!

please check:
 -  build the vigettes, and run `R CMD check` to ensure the examples are runnable.
 - add your name to DESCRIPTION (optional, but highly appreciated)

cbielow committed:
Apply suggestions from code review

Commit:
Updated DESCRIPTION, modified some text in docs

Commit:
Updated version number and release date;
Added documentation aliases to all classes in mzQC.R to handle documentation of their S4 counterparts;
Replaced snapshot test in test_io.R with a `verify_output()` test (solves false positives caused by some device- or environment-specific differences).

Commit:
Corrected outdated comment in mzQc.R

cbielow committed:
* ran pgkdown::build_site(); * augmented news.md

cbielow committed:
allow CLONE, to ease usage in downstream packages

Jeremi-Maciejewski committed:
Merge pull request #1 from MS-Quality-Hub/feat_fastRead
